app/models/infra_conversion_job.rb

```diff
@@ -0,0 +1,167 @@
+  #     |                         |                :start_post_stage       |
+  #     | :abort_job              | :abort_job                             |
+  #     \------------------------>|                                        | :finish
+  #                               v                                        |
+  #                             aborting --------------------------------->|
+  #                                                    :finish             v
+  #                                                                    finished
+  #
+
+  alias_method :initializing, :dispatch_start
+  alias_method :finish,       :process_finished
+  alias_method :abort_job,    :process_abort
+  alias_method :cancel,       :process_cancel
+  alias_method :error,        :process_error
+
+  def load_transitions
+    self.state ||= 'initialize'
+
+    {
+      :initializing     => {'initialize'       => 'waiting_to_start'},
+      :start            => {'waiting_to_start' => 'running'},
+      :poll_conversion  => {'running'          => 'running'},
+      :start_post_stage => {'running'          => 'post_conversion'},
+      :poll_post_stage  => {'post_conversion'  => 'post_conversion'},
+      :finish           => {'*'                => 'finished'},
+      :abort_job        => {'*'                => 'aborting'},
+      :cancel           => {'*'                => 'canceling'},
+      :error            => {'*'                => '*'}
+    }
+  end
+
+  def migration_task
+    @migration_task ||= target_entity
+    # valid states: %w(migrated pending finished active queued)
+  end
+
+  def start
+    # TransformationCleanup 3 things:
+    #  - kill v2v: ignored because no converion_host is there yet in the original automate-based logic
+    #  - power_on: ignored
+    #  - check_power_on: ignore
+
+    migration_task.preflight_check
+    _log.info(prep_message("Preflight check passed, task.state=#{migration_task.state}. continue ..."))
+    queue_signal(:poll_conversion)
+  rescue => error
+    message = prep_message("Preflight check has failed: #{error}")
+    _log.info(message)
+    abort_conversion(message, 'error')
+  end
+
+  def abort_conversion(message, status)
+    migration_task.cancel
+    queue_signal(:abort_job, message, status)
+  end
+
+  def polling_timeout(poll_type)
+    count = "#{poll_type.to_s}_count".to_sym
+    max = "#{poll_type.to_s}_max".to_sym
+    context[count] = (context[count] || 0) + 1
+    context[count] > options[max]
+  end
+
+  def poll_conversion
+    return abort_conversion("Polling times out", 'error') if polling_timeout(:poll_conversion)
+
+    message = "Getting conversion state"
+    _log.info(prep_message(message))
+
+    if migration_task.options[:virtv2v_wrapper].nil? || migration_task.options[:virtv2v_wrapper]['state_file'].nil?
+      message = "options[:virtv2v_wrapper]['state_file'] not available, continuing poll_conversion"
```

Comment:
The exact code that is nil probably isn't very helpful to users, maybe `Virt v2v state file not available`